feat: add release notes for 26.04 LTS (#201)
canonical/landscape-documentation: docs/reference/release-notes/26-04-lts-release-notes.md

---
myst:
  html_meta:
    description: "Release notes for Landscape 26.04 LTS (April 2026). Introduces security event logging, FDE recovery key management, release upgrade REST API, the new debarchive repository service, and multiple security fixes."
---

(reference-release-notes-26-04-lts)=
# 26.04 LTS release notes

> 

Reviewer: For this whole section before the highlights - you should pull it out of the note block. The bullet points are fine, just the note block looks too big when it should just be a summary at the top.

And before the details, add at the top line "You can now access self-hosted Landscape 26.04 LTS in our latest LTS PPA: `ppa:landscape/self-hosted-26.04`." (assuming that's the correct PPA name which it should be)

--

And also - the outbox snap and charm stuff should be represented outside the intro in the notes - presumably in the Upgrades section? See: https://documentation.ubuntu.com/landscape/reference/release-notes/25.10-release-notes/#upgrade
(i.e., you also need to add an upgrade section - The intro line can be short, "See our 26.04 upgrade guide for detailed steps.")

Reviewer: Oh also, in that Upgrade section, below the link to the upgrade guide, add a line that says "If you use repository management in Landscape, we recommend waiting to upgrade until the 26.04.1 point release (expected August 2026). There's currently a safeguard in place for 24.04 to 26.04 upgrades to block automatic upgrades for repository management users."

> 

Reviewer: Quickstart won't run on resolute, right? It's only for noble and jammy? (if yes, you can add  that to your list of notes, e.g., "The Landscape Server Quickstart package only runs on .."

<!-- TODO: 
> See also: {ref}`how-to-upgrade-to-26-04-lts`
-->
```{note}
- Landscape Server 26.04 LTS runs on Ubuntu 26.04 LTS Resolute Raccoon, 24.04 LTS Noble Numbat, or 22.04 LTS Jammy Jellyfish.
- Database schema changes are required to upgrade to Landscape Server 26.04 LTS.
- Landscape Server 26.04 LTS requires the `landscape-outbox` snap.
- The updated Landscape Server Charm will not be released until the Landscape Server 26.04.1 LTS point release in August 2026.
```

## Highlights

- **Security event logging**: Security-relevant actions are now recorded in the {ref}`reference-logs`, including:

  - Authentication success and failure events
  - Password reset events
  - JWT issuance, validation, and expiry events
  - User, employee, and admin profile updates
  - Role and permission changes
  - Authorization failures

> - **FDE recovery key management**: The server now provides REST API endpoints for Full Disk Encryption (FDE) recovery key management, enabling automated storage and retrieval of recovery keys for enrolled machines. A new client plugin (fde-recovery-key-manager) enables the client to securely report FDE recovery keys to the server.

Reviewer: ```suggestion
- **FDE recovery key management**: The server now provides REST API endpoints for Full Disk Encryption (FDE) recovery key management, enabling automated storage and retrieval of recovery keys for enrolled machines. A new client plugin (`fde-recovery-key-manager`) enables the client to securely report FDE recovery keys to the server.
```

Also: It's not just the API though - it's in the UI too

> - **Soft deletion of computers**: Computer deletion now happens asynchronously, allowing for a smoother user workflow.

Reviewer: Can you expand on this? What is the "smoother user workflow"?

Author: I wish I could. @srunde3 Is soft-deletion part of this release, or is it coming later?

Reviewer 2: it is part of this release. It'll be enabled by default for self-hosted, and we'll gradually implement on SaaS to monitor performance before enabling for all accounts.

Wording-wise. I'd say "computer records are cleaned up asynchronously, allowing for higher deletion throughput and fixing the time out errors that some users experience when deleting old computers" or something along those lines. The user impact is that they can delete more instances, especially in bulk.

> - **Release upgrade REST API**: New REST API endpoints for managing OS release upgrades have been added, providing improved control and status reporting compared to the legacy endpoints.

Reviewer: Also anything to add here? It's vague

- **Repository management improvements**:

  - Introduces the new `debarchive` service for repository management.

>   - Repository profiles have been updated with improved workflows.

Reviewer: What is the improved workflow?

Author: @Perfect5th Any suggestions here?

>   - A feature flag has been added for repository mirroring, with a safeguard that prevents upgrading from a version with existing repository mirrors until the feature is re-enabled.

Reviewer: We don't say "feature flag" in the docs / user communications - can you reframe this in a more user friendly way?

Although this could probably just be removed here in favor of the upgrade note - see my comment above

>   - The quickstart installer now includes `debarchive` for local archive support.

Reviewer: Does this need to be added though? This is the local file storage option (instead of S3), right?

  - The legacy API endpoints for repository management have been removed; the new archive management system replaces them.

> - **Ubuntu Installer attach service**: The Ubuntu Installer attach service is now included in the quickstart installation by default, allowing machines to be attached to Landscape during the Ubuntu installation process.

Reviewer: So (if I read it correctly) the only change for 26.04 is that it's been added to quickstart? I.e., no notable functionality change?

Author: @srunde3 can you confirm?

Reviewer 2: Thanks for the ping, I missed this on first review. I don't recall including it in quickstart, and the control file doesn't suggest that it is. I think this should be removed.

- **TLS support for RabbitMQ**: Standard TLS connections are now supported for RabbitMQ.

> 

Reviewer: There were a lot of charm changes too, right..? On the dependencies and stuff with pgbouncer?

Author: There are plenty of charm changes. But we aren't releasing the new charm until August, so I omitted the details.

- **REST API strict content-type enforcement**: HTTP `POST`, `PUT`, and `PATCH` requests to the v2 REST API now require the `Content-Type: application/json` request header. Requests that omit this header will receive a 415 Unsupported Media Type response.

- **Python version and dependency updates**:

  - Landscape Server now vendors its Python dependencies into a virtual environment and requires Python 3.12.
  - Dependencies on `netaddr`, `python3-bs4`, `python3-oops-amqp`, `python3-stripe`, `simplejson`, and `wkhtmltopdf` have been removed.
  - The build system has migrated from `setup.py/requirements.txt` to `pyproject.toml` with `uv`.

> - **Landscape Client - Configurable apt update timeout**: A new configuration option allows administrators to set a custom timeout for apt update operations, addressing issues with slow or unreliable package mirrors.

Reviewer: ```suggestion
- **Landscape Client - Configurable `apt update` timeout**: A new configuration option allows administrators to set a custom timeout for `apt update` operations, addressing issues with slow or unreliable package mirrors.
```

> 

Reviewer: You can have a "Highlights" and a "Additional updates" section - if you want to pull some of these out and just list them below, like here https://documentation.ubuntu.com/landscape/reference/release-notes/25.10-release-notes/#additional-updates

I don't have a strong preference if you split it or keep them all under "Highlights", but in general I think the Highlights could be loosely ordered based on user impact (soft deletion and release upgrade API could be moved down a bit IMO, although I can still see at least soft deletion as a "highlight"). The repository management one is a major change - I'd move it higher either at the top or below the security ones

---
myst:
  html_meta:
    description: "Release notes for Landscape 26.04 LTS (April 2026). Introduces security event logging, FDE recovery key management, release upgrade REST API, the new debarchive repository service, and multiple security fixes."
---

(reference-release-notes-26-04-lts)=
# 26.04 LTS release notes

<!-- TODO: 
> See also: {ref}`how-to-upgrade-to-26-04-lts`
-->
- Landscape Server 26.04 LTS runs on Ubuntu 26.04 LTS Resolute Raccoon, 24.04 LTS Noble Numbat, or 22.04 LTS Jammy Jellyfish. The Landscape Server Quickstart package only runs on 24.04 LTS Noble Numbat and 22.04 LTS Jammy Jellyfish.
- Database schema changes are required to upgrade to Landscape Server 26.04 LTS.

> You can now access Landscape 26.04 LTS in our 26.04 LTS PPA: `ppa:landscape/self-hosted-26.04`.

Reviewer: ```suggestion
You can now access Landscape 26.04 LTS in our 26.04 LTS PPA: `ppa:landscape/self-hosted-26.04`.

- Landscape Server 26.04 LTS runs on Ubuntu 26.04 LTS Resolute Raccoon, 24.04 LTS Noble Numbat, or 22.04 LTS Jammy Jellyfish. The Landscape Server Quickstart package only runs on 24.04 LTS Noble Numbat and 22.04 LTS Jammy Jellyfish.
- Database schema changes are required to upgrade to Landscape Server 26.04 LTS.
```
Recommending it's formatted this way mostly so it doesn't start with bullet points

## Highlights

- **Repository management improvements**:

>   - Introduces the new `debarchive` service for repository management.

Reviewer 3: Not sure if we should add a beta / edge warning here, since I believe the full documentation won't be ready until the point release

  - Repository profiles have been updated with improved workflows.
  - The legacy API endpoints for repository management have been removed; the new archive management system replaces them.

- **Security event logging**: Security-relevant actions are now recorded in the {ref}`reference-logs`, including:

  - Authentication success and failure events
  - Password reset events
  - JWT issuance, validation, and expiry events
  - User, employee, and admin profile updates
  - Role and permission changes
  - Authorization failures

- **FDE recovery key management**: The server now provides Full Disk Encryption (FDE) recovery key management, enabling automated storage and retrieval of recovery keys for enrolled machines. A new client plugin (`fde-recovery-key-manager`) enables the client to securely report FDE recovery keys to the server.

## Breaking changes

> - **Legacy API repository management endpoints removed**: Users relying on these endpoints should migrate to the new repository management API.

Reviewer: ```suggestion
- **Legacy API repository management endpoints removed**: Users relying on these endpoints will need to migrate to the new repository management API. See [Upgrades](#upgrades) for more details.
```
Adding this because we don't currently have docs for these users, so it feels odd to be like "hey you should migrate" w/o guidance, but in the Upgrades section we do guide people to wait.

## Additional updates

- **TLS support for RabbitMQ**: Standard TLS connections are now supported for RabbitMQ.

- **Landscape Client - Configurable `apt update` timeout**: A new configuration option allows administrators to set a custom timeout for `apt update` operations, addressing issues with slow or unreliable package mirrors.

- **Manage release upgrades**: You can now trigger release upgrades at scale from the web portal.

- **Soft deletion of computers**: Computer records are cleaned up asynchronously, allowing for higher deletion throughput and fixing the timeout errors that some users experience when deleting old computers.

- **Python version and dependency updates**:

  - Landscape Server now vendors its Python dependencies into a virtual environment and requires Python 3.12.
  - Dependencies on `netaddr`, `python3-bs4`, `python3-oops-amqp`, `python3-stripe`, `simplejson`, and `wkhtmltopdf` have been removed.
  - The build system has migrated from `setup.py/requirements.txt` to `pyproject.toml` with `uv`.

## Bug fixes

> - Fixed package changer not honouring the configured proxy when downloading packages.

Reviewer: ```suggestion
- Fixed package changer not honoring the configured proxy when downloading packages.
```
Now I'm doubting how our US spellchecker works

## Security fixes

Multiple security fixes are included in this release:

- Fixed user enumeration on the password reset page.
- Fixed a stored cross-site scripting (XSS) vulnerability on the activity result page.
- Fixed a stored XSS vulnerability in the profile creation page.
- Fixed HTML escaping in Legacy UI page templates.
- Fixed sanitization of access group titles and descriptions (both in the UI and Legacy API).
- Fixed an arbitrary file deletion vulnerability in the package upload handler.
- Fixed a cross-site request forgery (CSRF) vector via GET-based state-modifying requests.
- Fixed a JavaScript execution (XSS) vulnerability.
- Replaced `xml.etree` with `defusedxml` throughout the codebase.
- Fixed a Landscape Client issue where data file permissions were too permissive. [LP: #2121558](https://bugs.launchpad.net/landscape-client/+bug/2121558)
- Fixed content-type header validation for REST API endpoints.

## Supported third-party services

| Service | Compatible versions |
| ------- | ------------------- |
| PostgreSQL | 14, 16, 18 |
| HAProxy | 2.4, 2.8, 3.2 |
| RabbitMQ | 3.9, 3.12 |

## Upgrades

> See our 26.04 upgrade guide for detailed steps.

Reviewer 3: add link to upgrade guide if available

Reviewer: Agreed - you can add the hidden TODO part like you have at the top

> If you use repository management in Landscape, we recommend waiting to upgrade until the 26.04.1 point release (expected August 2026). There's currently a safeguard in place to block automatic upgrades to 26.04 LTS for repository management users.

Reviewer 3: Not sure if it's worth mentioning reprepo specifically, or if that's pretty much all users at this point

Reviewer: IMO I wouldn't mention it just because it's not mentioned in general in our docs and the package isn't that visible to our users. All repo management users use reprepro in pre-26.04 versions, so it's sufficient as is

---
myst:
  html_meta:
    description: "Release notes for Landscape 26.04 LTS (April 2026). Introduces security event logging, FDE recovery key management, release upgrade REST API, the new debarchive repository service, and multiple security fixes."
---

(reference-release-notes-26-04-lts)=
# 26.04 LTS release notes

<!-- TODO: 
> See also: {ref}`how-to-upgrade-to-26-04-lts`
-->
- Landscape Server 26.04 LTS runs on Ubuntu 26.04 LTS Resolute Raccoon, 24.04 LTS Noble Numbat, or 22.04 LTS Jammy Jellyfish. The Landscape Server Quickstart package only runs on 24.04 LTS Noble Numbat and 22.04 LTS Jammy Jellyfish.
- Database schema changes are required to upgrade to Landscape Server 26.04 LTS.

You can now access Landscape 26.04 LTS in our 26.04 LTS PPA: `ppa:landscape/self-hosted-26.04`.

## Highlights

- **Repository management improvements**:

  - Introduces the new `debarchive` service for repository management.
  - Repository profiles have been updated with improved workflows.

>   - The quickstart installer now includes `debarchive` for local archive support.

Reviewer 4: We went back on this due to the snap vs deb packaging issue, so quickstart does _not_ include debarchive. Instead we have #202 for instruction how to install debarchive.

---
myst:
  html_meta:
    description: "Release notes for Landscape 26.04 LTS (April 2026). Introduces security event logging, FDE recovery key management, release upgrade REST API, the new debarchive repository service, and multiple security fixes."
---

(reference-release-notes-26-04-lts)=
# 26.04 LTS release notes

> > See also: {ref}`how-to-upgrade-to-26-04-lts`

Author: It will be there when this is merged.

---
myst:
  html_meta:
    description: "Release notes for Landscape 26.04 LTS (April 2026). Introduces security event logging, FDE recovery key management, release upgrade REST API, the new debarchive repository service, and multiple security fixes."
---

(reference-release-notes-26-04-lts)=
# 26.04 LTS release notes

<!-- TODO: 
> See also: {ref}`how-to-upgrade-to-26-04-lts`
-->
```{note}
- Landscape Server 26.04 LTS runs on Ubuntu 26.04 LTS Resolute Raccoon, 24.04 LTS Noble Numbat, or 22.04 LTS Jammy Jellyfish.
- Database schema changes are required to upgrade to Landscape Server 26.04 LTS.
- Landscape Server 26.04 LTS requires the `landscape-outbox` snap.

> - The updated Landscape Server Charm will not be released until the Landscape Server 26.04.1 LTS point release in August 2026.

Reviewer 2: Nothing against this wording. I'll add that it will be in beta immediately, but perhaps we're deliberately omitting that in the release notes. It won't be in stable until the point release, as noted.

Author: Yeah, I was going to leave it out entirely. We can update the release notes then to describe the changes to the charm.

- **Ubuntu Installer attach service**: The Ubuntu Installer attach service is now included in the quickstart installation by default, allowing machines to be attached to Landscape during the Ubuntu installation process.

- **TLS support for RabbitMQ**: Standard TLS connections are now supported for RabbitMQ.

- **REST API strict content-type enforcement**: HTTP `POST`, `PUT`, and `PATCH` requests to the v2 REST API now require the `Content-Type: application/json` request header. Requests that omit this header will receive a 415 Unsupported Media Type response.

- **Python version and dependency updates**:

  - Landscape Server now vendors its Python dependencies into a virtual environment and requires Python 3.12.
  - Dependencies on `netaddr`, `python3-bs4`, `python3-oops-amqp`, `python3-stripe`, `simplejson`, and `wkhtmltopdf` have been removed.
  - The build system has migrated from `setup.py/requirements.txt` to `pyproject.toml` with `uv`.

- **Landscape Client - Configurable apt update timeout**: A new configuration option allows administrators to set a custom timeout for apt update operations, addressing issues with slow or unreliable package mirrors.

## Breaking changes

> - **Legacy API repository management endpoints removed**: Users relying on these endpoints should migrate to the new repository management API.

Reviewer 2: Do we need to mention the `Content-Type: application/json` change here?

Author: I'll move the item from "Highlights" to "Breaking changes"
